### DISCUSSION TITLE
myloader is working very slow

### DISCUSSION THREAD

**Discussion by sanjay92-gupta** [mydumper/mydumper#1127] [Q&A]
Hi,
mydumper dumped data in about 1 hr or so but myloader seems to be very slow. I can see connections in mysql workbench and I see sessions doing insert and that table has over 300M rows and it has done 35 Million in 30 mins.

**Comment by sanjay92-gupta** [comment]
--innodb-optimize-keys                Creates the table without the indexes and it adds them at the end. Options: AFTER_IMPORT_PER_TABLE and AFTER_IMPORT_ALL_TABLES. Default: AFTER_IMPORT_PER_TABLE

I see default is AFTER_IMPORT_PER_TABLE, I see my one table is getting imported and I can see all indexes.
I see create table statement with Primary Key and KEY statement. 
Is this reason my import is running slow.

**Comment by davidducos** [comment]
Hi @sanjay92-gupta, 
if you use --innodb-optimize-keys, will be use the default. If you don't use --innodb-optimize-keys it will create the table as it is defined, which means that it will not create the index after import the data, as it will create the table with the indexes.

**Reply by sanjay92-gupta** [reply]
Thanks David
I was able to use mydumper and myloader to load about 300GB data . Mydumper was very quick and it took less than 25 mins, while myloader took 3 hrs and 20 mins
There were few issues since I am doing from AWS RDS to AWS RDS so some procedures didn't get imported and later on found out that those procedure have SET DEFINER= clause so may be that was reason since even creating then manually gave me error 
ERROR 1227 (42000): Access denied; you need (at least one of) the SUPER or SET_USER_ID privilege(s) for this operation
Database changed
So after removing SET DEFINER=, no errors

I have one MySQL RDS instance ( Source) and Created new MySQL RDS target instance and used Mydumper and myloader to load data and setup replication from source and target. All went well except some procedures got missed as explained above and one table in target is having more rows than source so not sure why more rows on target. This table does not have Primary key so may be this could be issue . Any idea why this could happen

**Reply by davidducos** [reply]
you have the option --skip-definer on mydumper and myloader to avoid the set definer issue. 

Please, share a test case of the table with more rows so we can troubleshoot it
